src/app/Console/Commands/RequireDevTools.php

```diff
@@ -135,27 +141,93 @@ public function handle()
+                $details = json_decode($buffer);
+            } elseif (File::exists('auth.json')) {
+                $details = json_decode(File::get('auth.json'), true)['http-basic']['backpackforlaravel.com'] ?? false;
+            }
             $this->progressBar->advance();
         });
 
-        // Finish
-        $this->progressBar->finish();
-        $this->info(' DevTools is now required.');
+        // Create an auth.json file
+        if (! $details) {
+            $this->info(' Creating auth.json file with your authentication token');
 
-        // DevTools inside installer
-        $this->info('');
-        $this->info(' Now running the DevTools installation command.');
+            $this->line(' (Find your access token details on https://backpackforlaravel.com/user/tokens)');
+            $username = $this->ask('Access token username');
+            $password = $this->ask('Access token password');
 
-        // manually include the command in the run-time
-        if (! class_exists(\Backpack\DevTools\Console\Commands\InstallDevTools::class)) {
-            include base_path('vendor/backpack/devtools/src/Console/Commands/InstallDevTools.php');
+            $process = new Process(['composer', 'config', 'http-basic.backpackforlaravel.com', $username, $password]);
+            $process->run(function ($type, $buffer) use ($username, $password) {
+                if ($type === Process::ERR) {
+                    // Fallback
+                    $authFile = [
+                        'http-basic' => [
+                            'backpackforlaravel.com' => [
+                                'username' => $username,
+                                'password' => $password,
+                            ],
+                        ],
+                    ];
+
+                    if (File::exists('auth.json')) {
+                        $currentFile = json_decode(File::get('auth.json'), true);
+                        if (! ($currentFile['http-basic']['backpackforlaravel.com'] ?? false)) {
+                            $authFile = array_merge_recursive($authFile, $currentFile);
+                        }
+                    }
+
+                    File::put('auth.json', json_encode($authFile, JSON_PRETTY_PRINT));
+                }
+
+                $this->progressBar->advance();
+            });
         }
+    }
 
-        $this->call(\Backpack\DevTools\Console\Commands\InstallDevTools::class);
+    public function requireDevTools()
+    {
+        // Require package
+        $process = new Process(['composer', 'require', '--dev', '--with-all-dependencies', 'backpack/devtools']);
+        $process->setTimeout(300);
+        $process->run(function ($type, $buffer) use ($process) {
+            if ($type === Process::ERR) {
+                // create a log channel
+                $log = new Logger('devtools_errors');
+                $log->pushHandler(new StreamHandler(storage_path('logs/devtools.log'), Logger::ERROR));
```

Comment:
Does this new `devtools.log` file show up in LogManager too?

Reply:
Yes it does.
![image](https://user-images.githubusercontent.com/7188159/139880099-8b7a96ff-543b-4b26-bb92-0b6c0127c116.png)

You can download it. But you cannot open it in browser:
![image](https://user-images.githubusercontent.com/7188159/139880201-4175990e-e149-4788-be31-2a9e4dc3aea2.png)

I've been trying to get down into this, all seems to go fine with `setFile` but then the `::all()` returns empty for this log file in particular. Maybe permissions or something ? Dunno yet.

I will investigate more if we decide to keep this log system.

Reply:
Yeah... don't spend too much time on this. I think it's better to use the normal log file (if we do log anything), no use in complicating things.